### ISSUE TITLE
Antialias cone gradients

### ISSUE THREAD

**Issue by humdingerb** [HaikuArchives/ArtPaint#572]
Conic gradients would look nicer if the border where colour 2 (white) encounters colour 1 (blue) again. See this zoomed-in example:

![cone](https://github.com/HaikuArchives/ArtPaint/assets/6107158/fd4bc0aa-720f-4c0f-98d1-b33233f41643)

**Comment by dsizzle** [comment]
I was going to say that the use-cases I’ve seen would mostly be where the start and end color are the same, but recently I saw where people make purposely hard edged conic gradients for things like pie charts, and so you’d want and hard edge to be antialiased.  

Not sure how hard it is to do, maybe if the previous pixel differs by more than a certain amount, add a pixel with a color halfway between? Or the average of the pixel to the left and above?

**Comment by humdingerb** [comment]
> I was going to say that the use-cases I’ve seen would mostly be where the start and end color are the same.

You mean doing the cone like WonderBrush:

![WBconic](https://github.com/HaikuArchives/ArtPaint/assets/6107158/999ce57e-fd96-422f-bc0c-27ebca8aba04)

That would solve our issue. Removes some artsy potential though...

**Comment by dsizzle** [comment]
no, that seems like it reverses the colors halfway thru - I mean once ArtPaint allows setting more colors in the gradient, the user would manually set the start and end to the same.  That's the use case I'm most familiar with.  For drawing stuff such as:

![ax-anisotropic-brushed-pot](https://github.com/HaikuArchives/ArtPaint/assets/10878750/90440eb9-4f92-4793-a113-acc2f65d2c86)

**Comment by humdingerb** [comment]
Right. It's more flexible (once more than two colours are possible), but needs to have the problem solved of antialiasing a hard colour switch, if the user sets the start and end colours to (very) different colours.
